### ISSUE TITLE
One shot training data annotation

### ISSUE THREAD

**Issue by haomengz** [Reagan1311/OOAL#7]
Hi,

In your paper you mentioned:

> AGD20K uses sparse annotation, where ground truth consists of keypoints within affordance areas, and then a gaussian kernel is applied over each point to produce dense annotation.

I am wondering whether you could provide some details about how you manually label the one-shot training data for AGD20K? Thanks!

**Comment by Reagan1311** [comment]
Hi, we first annoate sparse points with mouse:

```
image = cv2.imread('knife.jpg')
cv2.imshow('output', image)

# Function to capture mouse events
def get_coordinates(event, x, y, flags, param):
    if event == cv2.EVENT_LBUTTONDOWN:
        print(f"Coordinates: ({x}, {y})")

# Set the mouse callback function
cv2.setMouseCallback('output', get_coordinates)
cv2.waitKey(0)
cv2.destroyAllWindows()
```

Then, we apply a gaussian kernel over these points to get heatmaps:
```
h, w, _ = image.shape
k_ratio = 3
k_size = int(np.sqrt(h * w) / k_ratio)
if k_size % 2 == 0:
    k_size += 1

heatmap = np.zeros((h, w), dtype=np.float32)
points = [[985, 339], [1073, 280], [1162, 218]]

for point in points:
    x = point[1]
    y = point[0]
    row = int(x)
    col = int(y)
    heatmap[row, col] += 1.0

heatmap = cv2.GaussianBlur(heatmap, (k_size, k_size), 0)
heatmap = cv2.resize(heatmap,(224,224))
heatmap = (heatmap - np.min(heatmap)) / (np.max(heatmap) - np.min(heatmap) + 1e-10)
```

The is similar with the annotation pipeline in [Interaction Hotspots](https://openaccess.thecvf.com/content_ICCV_2019/supplemental/Nagarajan_Grounded_Human-Object_Interaction_ICCV_2019_supplemental.pdf) (see section S3)

**Comment by haomengz** [comment]
Thanks for the reply and sample code. 
May I know around how many key points you label for each image?

**Comment by Reagan1311** [comment]
The number of key points is not fixed (normally around 5 points for each object part). You can visualize the heatmap to check if it looks good.

**Comment by haomengz** [comment]
Thanks a lot for the help!
